<launch>
    <arg name="eye_on_hand" default="false" doc="eye-on-hand instead of eye_on_base" />
    <arg name="namespace_prefix" default="handeyecalibration" />
    <arg if="$(arg eye_on_hand)" name="namespace" value="$(arg namespace_prefix)_eye_on_hand" />
    <arg unless="$(arg eye_on_hand)" name="namespace" value="$(arg namespace_prefix)_eye_on_base" />

    <!-- The input reference frames -->
    <arg name="base_link_frame"       default="base_link" />
    <arg name="tool_frame"            default="tool0" />
    <arg name="optical_origin_frame"  default="optical_origin" />
    <arg name="optical_target_frame"  default="optical_target" />

    <!-- Publish frames while calibrating -->
    <arg name="publish_tf" default="true"/>

    <!-- Run the ViSP solver to determine the extrinsic parameters -->
    <node ns="$(arg namespace)" name="hand_eye_solver"
          pkg="visp_hand2eye_calibration"
          type="visp_hand2eye_calibration_calibrator">
        <remap from="/compute_effector_camera_quick" to="compute_effector_camera_quick"/>
    </node>

    <group ns="$(arg namespace)" >
        <param name="eye_on_hand"           value="$(arg eye_on_hand)"/>
        <param name="base_link_frame"       value="$(arg base_link_frame)" />
        <param name="tool_frame"            value="$(arg tool_frame)" />
        <param name="optical_origin_frame"  value="$(arg optical_origin_frame)" />
        <param name="optical_target_frame"  value="$(arg optical_target_frame)" />
    </group>

    <!-- Connect the tracker to the solver -->
    <node name="hand_eye_connector"
          ns="$(arg namespace)"
          pkg="handeyecalibration"
          type="calibrate.py"
          output="screen" />

</launch>
